tools/toolchains/mbed_toolchain.py

```diff
@@ -1393,3 +1393,34 @@ def report(self):
         to_ret['linker'] = {'flags': copy(self.flags['ld'])}
         to_ret.update(self.config.report)
         return to_ret
+
+def replace_small_c_lib(target, toolchain, check_only=False):
+    """
+    Replace the small C lib for the standard C lib.
+
+    The replacement occurs only if the small C lib is not supported but the
+    standard C lib is.
+    Return True if the replacement occurs otherwise return None.
+    """
+    if (
+        not is_library_supported("small", target, toolchain)
+        and is_library_supported("std", target, toolchain)
+        and target.c_lib == "small"
+    ):
+        if not check_only:
+            target.c_lib = "std"
```

Comment:
Do this where the function is called, and remove the 'check_only' flag altogether.

Reply:
Did you not say up there to do all of this in the base class instead of repeating it for every toolchain?

Reply:
12704f855b1b881af440bf8b97b8fb21d32f0cc1

Reply:
Yes, like the change, and yes, the three repeated bits of code could be pushed up into the __init__ method on the base, but this looks ok now.